tests/test_opnieuw.py

```diff
@@ -65,9 +66,44 @@ def test_retry_times(self):
         except TypeError as e:
             end = time.time()
             t_diff = end - start
-            self.assertGreater(t_diff, 1)
+            self.assertGreater(t_diff, 0.5)
             self.assertEqual(self.counter, 3)
 
+    def test_retry_immediately_global(self) -> None:
+        start = time.time()
+        self.counter = 0
+        with retry_immediately():
+            try:
+                self.foo()
+            except TypeError as e:
+                end = time.time()
+                t_diff = end - start
+                self.assertLess(t_diff, 0.5)
```

Comment:
This assert is asserting as much about the performance of the machine that runs the test, and the willingness of the OS to not preempt and shelve the process in between the two `time()` calls, as it is asserting anything about retry behavior. It is probably not a problem in practice, but relying on the current time in tests is a recipe for flaky tests.

Your trick for swapping out the retry state in tests would also work for clocks, we could do a similar thing later to swap out `time.monotonic()` for a virtual clock to make the tests deterministic. But let's not do that in this pull request.